File: app/src/main/java/com/quran/labs/androidquran/ui/helpers/SingleAyahHighlight.java
package com.quran.labs.androidquran.ui.helpers;

import java.util.HashSet;
import java.util.Set;

public class SingleAyahHighlight extends AyahHighlight {

  public SingleAyahHighlight(String key) {
    super(key);
  }

  public SingleAyahHighlight(int surah, int ayah) {
    super(surah + ":" + ayah);
  }

  public static Set<AyahHighlight> createSet(Set<String> ayahKeys) {
    Set<AyahHighlight> set = new HashSet<>();
    for(String ayahKey: ayahKeys) {
      set.add(new SingleAyahHighlight(ayahKey));
    }
    return set;
  }
}

File: app/src/main/java/com/quran/labs/androidquran/ui/helpers/SingleAyahHighlight.kt
package com.quran.labs.androidquran.ui.helpers

class SingleAyahHighlight : AyahHighlight {

  constructor(key: String) : super(key)

  constructor(surah: Int, ayah: Int) : super("$surah:$ayah")

  companion object {
    @JvmStatic
    fun createSet(ayahKeys: Set<String>): Set<AyahHighlight> {
      return ayahKeys.map { SingleAyahHighlight(it) }.toSet()
    }
  }
}
